components/content-service/pkg/initializer/git.go

```diff
@@ -144,12 +144,14 @@ func (ws *GitInitializer) realizeCloneTarget(ctx context.Context) (err error) {
 	if ws.TargetMode == RemoteBranch {
 		// create local branch based on specific remote branch
 		if err := ws.Git(ctx, "checkout", "-B", ws.CloneTarget, "origin/"+ws.CloneTarget); err != nil {
-			log.WithField("remoteURI", ws.RemoteURI).WithField("branch", ws.CloneTarget).Debug("Remote branch doesn't exist.")
-			return nil
+			log.WithField("remoteURI", ws.RemoteURI).WithField("branch", ws.CloneTarget).Error("Remote branch doesn't exist.")
+			err = checkGitStatus(err)
```

Comment:
Not understand why we check git statu at here 🤔 
https://github.com/gitpod-io/gitpod/blob/7640f4baf33c6840e3f5739f60214622b7db9b3c/components/content-service/pkg/initializer/git.go#L186-L194

Reply:
My understanding is that any git checkout command can encounter that Cloudflare error code, so I think we should handle it in every case?

Reply:
Sounds reasonable. 💯